### ISSUE TITLE
Compiler doesn't support global keyword

### ISSUE THREAD

**Issue by ptliu** [ucsd-cse231-w21/chocopy-wasm-compiler#39]
In both ChocoPy and Python, use of the global keyword is required to mutate a global variable when in function/method scope. For example, the following program throws an error in both ChocoPy and Python:

```python
x:int = 1
def test2():
  x = x + 1 #complains about x being mutated
```

Currently, our compiler doesn't require this at all, and instead will happily accept the above program. Should we accept this difference in semantics? It makes testing quite tricky, since there are now valid programs for our compiler that are not valid ChocoPy programs.

**Comment by ptliu** [comment]
@yalhessi @jpolitz is this worth fixing?

**Comment by jpolitz** [comment]
Yeah, it's worth fixing (sorry for the slow response, was PR-processing more than issue processing)
